edge/pkg/metamanager/metaserver/kubernetes/storage/storage.go

```diff
@@ -114,6 +114,44 @@ func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions)
 	return obj, err
 }
 
+// PassThrough
+// The request is routed to the dynamic controller via the metaServer.
+// If the request is approved, the response will be saved to local storage.
+// It will be acquired from local data storage if it fails.
```

Comment:
I understand the local data storage is `fallback` behavior here.

my question here is we are expecting the version in k8s would be upgraded during runtime?

if not, can we do instead,

1. check the local storage if `GetPassThroughObj` exist?
2. and the if that does, return immediately with using local cache? and if not, try dynamic controller to pass through the request?

the reason above is, most likely version is the same, so it would be more cost effective not to access the dynamic controller every single time?

my assumption would be completely wrong since i am kinda new for KubeEdge. So i would like to hear the insights.

Reply:
It is not appropriate to use local storage directly. When the version of the kubernetes core component is upgraded, kubeedge does not recognize the upgrade and cannot refresh the storage.

Reply:
> When the version of the kubernetes core component is upgraded, kubeedge does not recognize the upgrade and cannot refresh the storage.

+1, now the kubernetes version will not push to edge if edge does't request it, so the local data storage as fallback is reasonable now, and welcome more discussion here :) @fujitatomoya